includes/managers/class-fs-debug-manager.php

```diff
@@ -0,0 +1,511 @@
+            }
+
+            if (!FS_Logger::is_storage_logging_on()) {
+                return;
+            }
+
+            $limit  = min( ! empty( $_POST['limit'] ) ? absint( $_POST['limit'] ) : 200, 200 );
+            $offset = min( ! empty( $_POST['offset'] ) ? absint( $_POST['offset'] ) : 200, 200 );
+
+            $logs = FS_Logger::load_db_logs(
+                fs_request_get( 'filters', false, 'post' ),
+                $limit,
+                $offset
+            );
+
+            Freemius::shoot_ajax_success( $logs );
+        }
+
+        /**
+         * @author Vova Feldman (@svovaf)
+         *  Moved from Freemius
+         *
+         * @since  1.2.1.7
+         */
+        static function _get_db_option() {
+            check_admin_referer( 'fs_get_db_option' );
+
+            $option_name = fs_request_get( 'option_name' );
+
+            if ( ! is_super_admin() ||
+                 ! fs_starts_with( $option_name, 'fs_' )
+            ) {
+                Freemius::shoot_ajax_failure();
+            }
+
+            $value = get_option( $option_name );
+
+            $result = array(
+                'name' => $option_name,
+            );
+
+            if ( false !== $value ) {
+                if ( ! is_string( $value ) ) {
+                    $value = json_encode( $value );
+                }
+
+                $result['value'] = $value;
+            }
+
+            Freemius::shoot_ajax_success( $result );
+        }
+
+        /**
+         * @author Vova Feldman (@svovaf)
+         *  Moved from Freemius
+         *
+         * @since  1.2.1.7
+         */
+        static function _set_db_option() {
+            check_admin_referer( 'fs_set_db_option' );
+
+            $option_name = fs_request_get( 'option_name' );
+
+            if ( ! is_super_admin() ||
+                 ! fs_starts_with( $option_name, 'fs_' )
+            ) {
+                Freemius::shoot_ajax_failure();
+            }
+
+            $option_value = fs_request_get_raw( 'option_value' );
+
+            if ( ! empty( $option_value ) ) {
+                update_option( $option_name, $option_value );
+            }
+
+            Freemius::shoot_ajax_success();
+        }
+
+        /**
+         * @author Vova Feldman (@svovaf)
+         *  Moved from Freemius
+         *
+         * @since  1.1.7.3
+         */
+        static function _toggle_debug_mode() {
+            check_admin_referer( 'fs_toggle_debug_mode' );
+
+            if ( ! is_super_admin() ) {
+                return;
+            }
+
+            $is_on = fs_request_get( 'is_on', false, 'post' );
+
+            if ( fs_request_is_post() && in_array( $is_on, array( 0, 1 ) ) ) {
+                if ( $is_on ) {
+                    self::_turn_on_debug_mode();
+                } else {
+                    self::_turn_off_debug_mode();
+                }
+
+                // Turn on/off storage logging.
+                FS_Logger::_set_storage_logging( ( 1 == $is_on ) );
```

Comment:
@DanieleAlessandra please see that you are already setting storage login from the `_turn_off_debug_mode` or the `_turn_on_debug_mode` above (line 284). So I feel this call is redundant.